ValveAnnulusAnalysis/HeartValveWidgets/ValveSequenceBrowserWidget.py

```diff
@@ -0,0 +1,267 @@
+    if not self.valveVolumeBrowserNode or self.valveVolumeBrowserNode.GetPlaybackActive():
+      self.disableGUI()
+      return
+
+    seqIndex, seqIndexValue = self.valveBrowser.getDisplayedValveVolumeSequenceIndexAndValue()
+    hasTimepoint = seqIndexValue and self.valveBrowser.heartValveSequenceNode.GetItemNumberFromIndexValue(seqIndexValue) >= 0
+
+    self.ui.setEnabled(self.valveBrowser.valveBrowserNode is not None if self.valveBrowser else False)
+    self.ui.removeTimePointButton.enabled = hasTimepoint
+    self.ui.cardiacCyclePhaseSelector.enabled = hasTimepoint
+
+    t = Template("$action volume (index = $index)")
+    if hasTimepoint:
+      self.ui.addTimePointButton.text = t.substitute(action="Go to", index=seqIndex + 1)
+      heartValveNode = self.valveBrowser.heartValveSequenceNode.GetDataNodeAtValue(seqIndexValue)
+      self._setCardiacCycleIndex(
+        self.ui.cardiacCyclePhaseSelector.findText(heartValveNode.GetAttribute("CardiacCyclePhase"))
+      )
+    else:
+      self.ui.addTimePointButton.text = t.substitute(action="Add", index=seqIndex + 1)
+      self._setCardiacCycleIndex(0)
+
+  def updateGUIFromHeartValveNode(self, unusedArg1=None, unusedArg2=None, unusedArg3=None):
+    self.ui.setEnabled(self.valveBrowser.valveBrowserNode is not None if self.valveBrowser else False)
+
+    self._setCardiacCycleIndex(
+      self.ui.cardiacCyclePhaseSelector.findText(self.valveModel.getCardiacCyclePhase()) if self.valveModel else 0
+    )
+
+    valveVolumeSequenceIndexStr = self.valveModel.getVolumeSequenceIndexAsDisplayedString(
+      self.valveModel.getValveVolumeSequenceIndex()) if self.valveModel else ""
+    self.ui.valveVolumeSequenceIndexValue.setText(valveVolumeSequenceIndexStr)
+
+    valveModelSelected = self.valveModel is not None
+    self.ui.cardiacCyclePhaseSelector.setEnabled(valveModelSelected)
+
+  def _setCardiacCycleIndex(self, index):
+    wasBlocked = self.ui.cardiacCyclePhaseSelector.blockSignals(True)
+    self.ui.cardiacCyclePhaseSelector.setCurrentIndex(index)
+    self.ui.cardiacCyclePhaseSelector.blockSignals(wasBlocked)
+
+  def onValveBrowserNodeModified(self, observer=None, eventid=None):
+    # Show current valve volume if switched valve time point
+    if self.valveBrowser and self.valveVolumeNode:
+      itemIndex, indexValue = self.valveBrowser.getDisplayedHeartValveSequenceIndexAndValue()
+      if indexValue is not None:
+        if self.lastValveBrowserSelectedItemIndex != itemIndex:
+          self.lastValveBrowserSelectedItemIndex = itemIndex
+          # Switch volume
+          volumeItemIndex = self.valveBrowser.volumeSequenceNode.GetItemNumberFromIndexValue(indexValue)
+          self.valveBrowser.volumeSequenceBrowserNode.SetSelectedItemNumber(volumeItemIndex)
+      else:
+        # No valve node yet in the sequence
+        self.lastValveBrowserSelectedItemIndex = -1
+    else:
+      self.lastValveBrowserSelectedItemIndex = -1
+
+    self.updateGUIFromHeartValveNode()
```

Comment:
It looks like the file does not end with a newline. I don't think we have a coding rule, but in general it is preferable to have a newline ending for all text files.